express/dataset.py

```diff
@@ -0,0 +1,209 @@
+"""This module defines the FondantDataset class, which is a wrapper around the manifest.
+
+It also defines the FondantComponent class, which uses the FondantDataset class to manipulate data.
+"""
+
+from abc import abstractmethod
+import argparse
+import json
+from pathlib import Path
+
+import dask.dataframe as dd
+
+from express.component_spec import ExpressComponent, kubeflow2python_type
+from express.manifest import Manifest, Subset, Index
+
+
+class FondantDataset:
+    """Wrapper around the manifest to download and upload data into a specific framework.
+
+    Uses Dask Dataframes for the moment.
+    """
+
+    def __init__(self, manifest):
+        self.manifest = manifest
+
+    @property
+    def project_name(self):
+        return self.manifest.project_name
+
+    def _load_subset(self, name: str) -> dd.DataFrame:
+        # get subset from the manifest
+        subset = self.manifest.get_subset(name)
+        # get its location and fields
+        # TODO remove gcp prefix
+        location = "gcs://" + subset.location + ".parquet"
+        fields = list(subset.fields.keys())
+
+        df = dd.read_parquet(
+            location, columns=fields, storage_options={"project": self.project_name}
```

Comment:
The columns to read here should be based on the `ComponentSpec` instead of the `Manifest`, right?

Reply:
Yes that's right, so what happens is we read the `input_subsets` from the `ComponentSpec`, which list which subsets to load. Next we get the corresponding subset from the manifest to get its location.

Reply:
Yes, but I'm talking about which columns of the subset to load

Reply:
Ok yes that's right, have addressed that!